src/routes/actions.js

```diff
@@ -42,7 +42,7 @@ router
     if (from) {
       actions = await Action.aggregate(criteria)
       actions.forEach(action => {
-        action.location = location
+        setDistance(action, location)
```

Comment:
J'aime l'idée d'être plus explicite que ca ne l'était.
La façon du muter l'object depuis l'intérieur de la fonction me fait un peut bizarre, surtout sur un _model_ qui a la possiblité de définir proprement l'ensemble de ses méthodes et propriétés.
Du coup je ne suis pas certain de comprendre la raison qui le fait sortir du _model_.

Je n'y suis pas accroché cependant…

Reply:
> Du coup je ne suis pas certain de comprendre la raison qui le fait sortir du model.

Principalement parce que : 

> ça ne marche pas avec mongoose. En effet, selon la façon dont on fait la requête, il est possible que mongoose n'applique pas les setters. C'est le cas par exemple quand une aggrégation est faite : les setters ne sont pas appliqués sur l'objet.

Alors je pourrais faire une fonction statique dans le modèle, mais du coup on ne pourrait pas se servir du `this` (car l'aggrégation (entre autre) délie du modèle). Et on risquerait d'oublier : avec une fonction à côté, on est plus explicite.

> sur un model qui a la possiblité de définir proprement l'ensemble de ses méthodes et propriétés.

Justement, non (sauf erreur de ma part), voir ci-dessus, c'est pas propre car ça ne se fait pas tout le temps. En tout cas il semble qu'on ne connaissait pas cet écueil, donc on risque de ne pas remarquer (et de ne pas savoir tous les cas).

Par ailleurs, le fait d'enrichir les données d'un dans le but de le retourner à un client n'est pas nécessairement lié au modèle, on pourrait même plutôt argumenter que c'est lié au contrôleur (route chez nous) : en soit ni l'acteur, ni l'action n'a de distance absolue. Mais la route `actions` avec certains paramètres peut renvoyer une distance. Et une ressource (au sens objet) n'est pas sensé avoir une propriété mouvante en fonction de paramètres passés en entrée d'une route. Il ne s'agit donc que d'un enrichissement d'une réponse (on évite aussi la duplication ou l'héritage).

> La façon du muter l'object depuis l'intérieur de la fonction me fait un peut bizarre

Moi aussi, ça m'embête même ! Vu qu'un objet est passé par ref en JS, en
gardant le code tel qu'il était avant ça modifiait l'objet. Mais comme ça nous embête tous les deux on pourrait faire : 

```
actor.distance = distance(actor, params.location)
```

@vinyll tu as unwatch le projet c'est peut être pour ça que tu ne vois plus les notfis ?

Reply:
J'ai les notifs, c'est juste que c'est noyé dans les masses de notifsss (désinformation par surinformation…)

ok pour `actor.distance = distance(actor, params.location)`, ca me semble plus explicite.